console/console-init/ui/src/modules/address/dialogs/CreateAddress/Configuration.tsx

```diff
@@ -0,0 +1,191 @@
+      longDescription: string;
+      shortDescription: string;
+    };
+  }>;
+}
+export const AddressDefinition: React.FunctionComponent<IAddressDefinition> = ({
+  addressspaceName,
+  namespace,
+  addressName,
+  addressSpacePlan,
+  handleAddressChange,
+  isNameValid,
+  type,
+  setType,
+  plan,
+  setPlan,
+  addressSpaceType,
+  topic,
+  setTopic,
+  typeOptions,
+  setTypeOptions,
+  topicsForSubscription,
+  setTopicForSubscripitons,
+  planOptions,
+  setPlanOptions
+}) => {
+  const [isTypeOpen, setIsTypeOpen] = useState<boolean>(false);
+  const [isTopicOpen, setIsTopicOpen] = useState<boolean>(false);
+  const client = useApolloClient();
+
+  const onTypeSelect = async (event: any) => {
+    if (
+      event.currentTarget.childNodes[0] &&
+      event.currentTarget.childNodes[0].value
+    ) {
+      const type = event.currentTarget.childNodes[0].value;
+      setType(type);
+      const addressPlans = await client.query<IAddressPlans>({
+        query: RETURN_ADDRESS_PLANS(addressSpacePlan, type)
+      });
+      if (addressPlans.data && addressPlans.data.addressPlans.length > 0) {
+        const planOptions = addressPlans.data.addressPlans.map(plan => {
+          return {
+            value: plan.metadata.name,
+            label: plan.spec.displayName || plan.metadata.name,
+            description: plan.spec.shortDescription || plan.spec.longDescription
+          };
+        });
+        setPlan(" ");
+        setTopic(" ");
+        setPlanOptions(planOptions);
+      }
+      if (type === "subscription") {
+        const topics_addresses = await client.query<IAddressResponse>({
+          query: RETURN_TOPIC_ADDRESSES_FOR_SUBSCRIPTION(
+            addressspaceName,
+            namespace,
+            type
+          )
+        });
+        if (
+          topics_addresses.data &&
+          topics_addresses.data.addresses &&
+          topics_addresses.data.addresses.addresses.length > 0
+        ) {
+          const topics = topics_addresses.data.addresses.addresses.map(
+            address => {
+              return {
+                value: address.spec.address,
+                label: address.metadata.name
+              };
+            }
+          );
+          setTopicForSubscripitons(topics);
+        }
+      }
+      setIsTypeOpen(!isTypeOpen);
+    }
+  };
+
+  const [isPlanOpen, setIsPlanOpen] = useState(false);
+  const onPlanSelect = (event: any) => {
+    event.currentTarget.childNodes[0] &&
+      setPlan(event.currentTarget.childNodes[0].value);
+    setIsPlanOpen(!isPlanOpen);
+  };
+  const onTopicSelect = (event: any) => {
+    event.currentTarget.childNodes[0] &&
+      setTopic(event.currentTarget.childNodes[0].value);
+    setIsTopicOpen(!isTopicOpen);
+  };
+  const { loading, error, data } = useQuery<IAddressTypes>(
+    RETURN_ADDRESS_TYPES,
+    {
+      variables: {
+        a: addressSpaceType
+      }
+    }
+  );
+  if (loading) return <Loading />;
+  if (error) return <Loading />;
```

Comment:
I don't think we need a condition for errors here anymore

Reply:
I will remove this in next PR.